source/unified-test-format/unified-test-format.rst

```diff
@@ -1798,6 +1885,92 @@ Use a ``listIndexes`` command to check whether the index exists. Note that it is
 currently not possible to run ``listIndexes`` from within a transaction.
 
 
+loop
+~~~~
+
+The ``loop`` operation executes sub-operations in a loop until a termination
+signal is received by the unified test runner. It supports the following
+arguments:
+
+- ``operations``: the sub-operations to run on each loop iteration.
+  Each sub-operation must be a valid operation as described in this
+  specification.
+
+- ``storeErrorsAsEntity``: if specified, the runner MUST handle errors
+  arising during sub-operation execution and append a document with error
+  information to the array stored in the specified entity. If
+  ``storeFailuresAsEntity`` is specified, the runner MUST NOT include
+  failures in the errors. If ``storeFailuresAsEntity`` is not specified,
+  the runner MUST include failures in the errors. The error document
+  MUST contain the following fields:
+  
+  - ``error``: the textual description of the error encountered.
+  - ``time``: the number of (floating-point) seconds since the Unix epoch
+    when the error was encountered.
+
+- ``storeFailuresAsEntity``: if specified, the runner MUST handle failures
+  arising during sub-operation execution and append a document with failure
+  information to the array stored in the specified entity. If
+  not specified, the runner MUST treat failures as errors, and either
+  handle them following the logic described in ``storeErrorsAsEntity``
+  or cause them to terminate execution, if ``storeErrorsAsEntity`` is not
+  specified. The failure document MUST contain the following fields:
+  
+  - ``error``: the textual description of the failure encountered.
+  - ``time``: the number of (floating-point) seconds since the Unix epoch
+    when the failure was encountered.
+
+- ``storeIterationsAsEntity``: if specfied, the runner MUST keep track of
+  the number of iterations of the loop performed, and store that number
+  in the specified entity.
+
+The following termination behavior MUST be implemented by the unified test
+runner:
+
+- There MUST be a way to request termination of the loops. This request
+  will be made by the Atlas testing workload executor in response to
+  receiving the termination signal from Astrolabe.
+  
+- When the termination request is received, the workload executor MUST
+  stop looping. The current loop iteration SHOULD complete to its natural
+  conclusion (success or failure).
+
+- After the termination request is received, the runner MUST NOT start
+  any further loops.
+  
+- When the termination request is received, the runner SHOULD skip
+  any non-loop operations not yet started and terminate as soon as practical.
+  
+- Termination request MUST cause a successful termination of the test
+  overall, i.e., receiving the termination request MUST NOT by itself be
+  considered an error.
+
+The exact mechanism by which the termination signal is delivered to the
+unified test runner, including the respective API, is left to the driver.
```

Comment:
Related to my earlier suggestion for accessing the entity map, I could imagine using callbacks to control loop execution. I'm not sure if that's something you'd want to discuss here, or just leave this open-ended and up to each implementation to figure out.

Reply:
This section has now been made more generic. I haven't made any prescriptions as far as the termination mechanism therefore I think it makes sense to not mention callbacks either.